pkg/registry/common/querycache/cache_entry.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package querycache
+
+import (
+	"sync"
+	"time"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+)
+
+type cacheEntry struct {
+	nse            *registry.NetworkServiceEndpoint
+	expireTimeout  time.Duration
+	expirationTime time.Time
+	timer          *time.Timer
```

Comment:
Why we cannot use one ticker to clean all outdated entries instead of have a timer for each entry?

Reply:
Nice catch, reworked to single ticker.